src/data/sym2.lean

```diff
@@ -451,6 +453,21 @@ begin
   rw other_eq_other',
 end
 
+lemma filter_image_quotient_mk [decidable_eq α] (s : finset α) :
+  ((s.product s).image quotient.mk).filter (λ a : sym2 α, ¬a.is_diag) =
+    s.off_diag.image quotient.mk :=
+begin
+  ext ⟨x, y⟩,
```

Comment:
It's longer, but the goal state ends up cleaner as
```suggestion
  ext z,
  induction z using quotient.induction_on,
  rcases z with ⟨x, y⟩,
```
(it doesn't mix `quot.mk` and `quotient.mk`)